packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -626,21 +637,29 @@ export class ElectronMainApplication {
 
     protected restart(id: number): void {
         this.restarting = true;
-        const window = BrowserWindow.fromId(id);
-        window.on('closed', async () => {
+        const browserWindow = BrowserWindow.fromId(id);
+        // eslint-disable-next-line no-null/no-null
+        if (browserWindow === null) {
+            throw new Error(`no BrowserWindow with id: ${id}`);
+        }
```

Comment:
Minor: We should be able to not use `null`
```suggestion
        if (!browserWindow) {
            throw new Error(`no BrowserWindow with id: ${id}`);
        }
```